keras/layers/core.py

```diff
@@ -56,7 +56,7 @@ def __init__(self, mask_value=0., **kwargs):
         self.mask_value = mask_value
 
     def compute_mask(self, inputs, mask=None):
-        return K.any(K.not_equal(inputs, self.mask_value), axis=-1)
+        return K.cast(K.any(K.not_equal(inputs, self.mask_value), axis=-1), 'bool')
```

Comment:
Are there cases where `any` would not return `bool`? If so, better to fix it in the corresponding backend.